source/beamline/BioXAS/BioXASSideBeamStatusControl.cpp

```diff
@@ -0,0 +1,185 @@
+#include "BioXASSideBeamStatusControl.h"
+#include "actions3/AMActionSupport.h"
+
+BioXASSideBeamStatusControl::BioXASSideBeamStatusControl(const QString &name, QObject *parent) :
+	BioXASBeamStatusControl(name, parent)
+{
+	// Initialize inherited variables.
+
+	setContextKnownDescription("SideBeamStatus");
+
+	// Initialize local variables.
+
+	frontEndStatus_ = 0;
```

Comment:
This control (BioXASSideBeamStatusControl), all three statuses have cascading effects on each other. The SOE beam status is kind of the termination point--if this control can't stop, it means that either the endstation shutter can't stop or the POE control can't stop. If the POE can't stop, it means that either the mirror mask can't stop, the mono mask can't stop, or the front end beam status can't stop. I could add all three statuses here, but I think just asking the SOE status control has the same effect.